plugins/ssh-plugin/src/git/git-listener.ts

```diff
@@ -8,106 +8,164 @@
-    try {
-      keys = await che.ssh.getAll('vcs');
-    } catch (e) {
-      console.error(e && e.message ? e.message : e);
-      keys = [];
+  private async onGitOutput(data: string): Promise<void> {
+    if (data.startsWith('> git clone') || data.startsWith('> git push')) {
+      this.gitCommand = data;
+      return;
     }
 
-    return keys;
+    if (data.indexOf('Permission denied (publickey).') < 0) {
+      // In this listener we handle only issues with SSH key
+      return;
+    }
+
+    // Parse Git log events.
+    if (this.gitCommand.startsWith('> git clone') || this.gitCommand.startsWith('> git push')) {
+      const split = this.gitCommand.split(' ');
+      const command = split[2];
+      const uri = split[3];
+      const path = split[4];
+
+      if (command === 'clone') {
+        await this.handleGitClone(uri, path);
+      } else if (command === 'push') {
+        await this.handleGitPush(uri, path);
+      }
+    }
   }
 
-  async init() {
-    const keys = await this.getKeys();
+  private isGitHubUri(uri: string): boolean {
+    return uri.startsWith('git@github.com');
+  }
 
-    let gitLogHandlerInitialized: boolean;
-    /* Git log handler, listens to Git events, catches the clone and push events.
-          Asks to Upload a public SSH key if needed before these operations.
-          Authenticates to Github if needed. */
-    const onChange = () => {
-      // Get the vscode Git plugin if the plugin is started.
-      const gitExtension = theia.plugins.getPlugin('vscode.git');
-      if (!gitLogHandlerInitialized && gitExtension && gitExtension.exports) {
-        // Set the initialized flag to true state, to not to initialize the handler again on plugin change event.
-        gitLogHandlerInitialized = true;
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const git: any = gitExtension.exports._model.git;
-        let command: string;
-        let url: string;
-        let path: string;
-        const listener = async (out: string) => {
-          // Parse Git log events.
-          const split = out.split(' ');
-          if (out.startsWith('> git clone') || out.startsWith('> git push')) {
-            command = split[2];
-            url = split[3];
-            path = split[4];
-            // Catch the remote access error.
-          } else if (out.indexOf('Permission denied (publickey).') > -1) {
-            // If the remote repository is a GitHub repository, ask to upload a public SSH key.
-            if ((await che.oAuth.isRegistered('github')) && out.indexOf('git@github.com') > -1) {
-              switch (command) {
-                case 'clone': {
-                  if (await this.addKeyToGitHub.run({ confirmMessage: MESSAGE_PERMISSION_DENIED_PUBLICKEY })) {
-                    await git.clone(url, path.substring(0, path.lastIndexOf('/')));
-                    theia.window.showInformationMessage(`Project ${url} successfully cloned to ${path}`);
```

Comment:
The notification about successful clone should be present as well

Reply:
Done